### PR TITLE
[CI] manylinux: WASI-NN Piper

### LINKED ISSUE #3576: [CI] manylinux: WASI-NN Piper
### Summary

Build and release WASI-NN Piper on `manylinux_2_28`

### Details

- [ ] Fix compilation errors
- [ ] Update `.github/workflows/matrix-extensions.json`

### Appendix

```
/opt/rh/gcc-toolset-13/root/usr/libexec/gcc/x86_64-redhat-linux/13/ld: src/ucd-tools/libucd.a(case.c.o): relocation R_X86_64_32S against .rodata' can not be used when making a shared object; recompile with -fPIC`
```

### PR DESCRIPTION
**Pull request by PeterD1524**
This PR will enable WASI-NN Piper manylinux builds.

closes #3576

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review agent](https://github.com/flows-network/github-pr-review/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Review by hydai — APPROVED** [review]
LGTM. @0yi0 Could you please take a look?

File: .github/workflows/reusable-build-extensions-on-manylinux.yml [hunk 1]
```diff
@@ -125,6 +125,11 @@ jobs:
         run: |
           # Running tests of wasm_bpf requires proper ebpf running environment
           mount -t debugfs none /sys/kernel/debug
+      - if: ${{ matrix.plugin == 'wasi_nn-piper' }}
+        name: Install dependencies
+        shell: bash
+        run: |
+          bash utils/wasi-nn/install-onnxruntime.sh
```
Comment (0yi0) [inline .github/workflows/reusable-build-extensions-on-manylinux.yml:132]: Could this be put in the [Dockerfile](https://github.com/WasmEdge/WasmEdge/blob/master/utils/docker/Dockerfile.manylinux_2_28-plugins-deps)?

Comment (0yi0) [reply .github/workflows/reusable-build-extensions-on-manylinux.yml:132]: p.s. Modification to dockerfiles can be a separate PR if CI needs an updated docker image.

Comment (PeterD1524) [reply .github/workflows/reusable-build-extensions-on-manylinux.yml:132]: @hydai We discussed this before at https://github.com/WasmEdge/WasmEdge/issues/3483#issuecomment-2262504175. WDYT? The current workflow passed. Is this enough to consider the behavior verified?

Comment (hydai) [reply .github/workflows/reusable-build-extensions-on-manylinux.yml:132]: Yup. The CI are all green. It's okay to add this script into dockerfiles now.

Comment (PeterD1524) [reply .github/workflows/reusable-build-extensions-on-manylinux.yml:132]: OK. I will move this into a separate PR. This PR will have to wait for the docker PR to be merged and the updated docker image.

Comment (PeterD1524) [reply .github/workflows/reusable-build-extensions-on-manylinux.yml:132]: docker PR: [#3614](https://www.github.com/WasmEdge/WasmEdge/pull/3614)

**Comment by hydai** [comment]
We should wait for the CI publish the new docker image and rerun this workflows:
https://github.com/WasmEdge/WasmEdge/actions/runs/10243906752

**Comment by hydai** [comment]
The docker images are published. Re-run CI jobs for verification.

**Review by hydai — APPROVED** [review]
LGTM

**Comment by hydai** [comment]
After this, I think we will have prebuilt assets at the release of 0.14.1. It would be great if we could also update the installer (v1/v2) then to support installing the WASI-NN Piper plugin.

**Pull request merged**